pkg/cli/dump.go

```diff
@@ -169,6 +183,17 @@ func runDump(cmd *cobra.Command, args []string) error {
 			}
 		}
 
+		if shouldDumpTypes && dumpCtx.dumpMode != dumpDataOnly {
+			if _, err := fmt.Fprintf(w, "SET experimental_enable_enums = true;\n"); err != nil {
```

Comment:
For cross-version compatibility, be careful to tolerate `dump` talking to a server prior to your changes being available.

Reply:
(I have commented elsewhere, including on #28948, that the current architecture is flawed - all this logic should be done server-side, in which case  the cross-version compat story is much easier.)